src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs

```diff
@@ -75,11 +75,19 @@ public override FileModel Load(FileAndType file, ImmutableDictionary<string, obj
 
             var filePath = Path.Combine(file.BaseDir, file.File);
             var repoDetail = GitUtility.GetGitDetail(filePath);
-            var displayLocalPath = repoDetail?.RelativePath ?? Path.Combine(file.BaseDir, file.File).ToDisplayPath();
+            string displayLocalPath = null;
+
+            object baseDirectory;
+            if (metadata.TryGetValue("baseRepositoryDirectory", out baseDirectory))
+            {
+                displayLocalPath = PathUtility.MakeRelativePath((string)baseDirectory, file.FullPath);
+            }
```

Comment:
to reduce duplicate code, how about move to SingleDocumentBuilder line 392, after load then set LocalPathFromRoot property?